extensions-contrib/druid-rocketmq/src/main/java/org/apache/druid/firehose/rocketmq/RocketMQFirehoseFactory.java

```diff
@@ -498,6 +502,7 @@ public void run()
       }
       catch (InterruptedException e) {
         LOGGER.error(e, "");
+        Thread.currentThread().interrupt();
```

Comment:
I think it's always better to put this statement first, before logging. Because if an exception is thrown from logging the interruption status is shallowed.

Reply:
I've updated my PR with a new commit to place such lines at first line at their corresponding code blocks.

Reply:
Let's assume that this PR is blocked by #6939.